lib/npg_pipeline/function/autoqc.pm

```diff
@@ -177,62 +183,80 @@ sub _create_definition_object {
 }
 
 sub _generate_command {
-  my ($self, $h, $is_multiplexed_lane) = @_;
+  my ($self, $dp) = @_;
 
   my $check     = $self->qc_to_run();
-  my $position  = $h->{'position'};
-  my $tag_index = $h->{'tag_index'};
-  my $c = sprintf '%s --check=%s --id_run=%i --position=%i',
-                  $QC_SCRIPT_NAME, $check, $h->{'id_run'}, $position;
-
-  if (defined $tag_index) {
-    $c .= q[ --tag_index=] . $tag_index;
-  }
+  my $archive_path = $self->archive_path;
+  my $recal_path= $self->recalibrated_path;
+  my $dp_archive_path = $dp->path($self->archive_path);
+  my $cache10k_path = $dp->short_files_cache_path($archive_path);
+  my $qc_out_path = $dp->qc_out_path($archive_path);
+  my $bamfile_path = File::Spec->catdir($dp_archive_path, $dp->file_name(ext => 'bam'));
+  my $fq1_filepath = File::Spec->catdir($cache10k_path, $dp->file_name(ext => 'fastq', suffix => '_1'));
+  my $fq2_filepath = File::Spec->catdir($cache10k_path, $dp->file_name(ext => 'fastq', suffix => '_2'));
+  my $fqt_filepath = File::Spec->catdir($cache10k_path, $dp->file_name(ext => 'fastq', suffix => '_t'));
+  my $fqc1_filepath = File::Spec->catdir($dp_archive_path, $dp->file_name(ext => 'fastqcheck', suffix => '1'));
+  my $fqc2_filepath = File::Spec->catdir($dp_archive_path, $dp->file_name(ext => 'fastqcheck', suffix => '2'));
+
+  my $c = sprintf '%s --check=%s --rpt_list="%s" --filename_root=%s --qc_out=%s',
+                  $QC_SCRIPT_NAME, $check, $dp->{rpt_list}, $dp->file_name_root, $qc_out_path;
 
   if ($check eq q[insert_size]) {
     $c .= $self->is_paired_read() ? q[ --is_paired_read] : q[ --no-is_paired_read];
   } elsif ($check eq q[qX_yield] && $self->platform_HiSeq) {
     $c .= q[ --platform_is_hiseq];
   }
 
-  my $qc_out = defined $tag_index ? $self->lane_qc_path($position) : $self->qc_path();
-  $qc_out or $self->logcroak('Failed to get qc_out directory');
-  my $qc_in  = defined $tag_index ? $self->lane_archive_path($position) : $self->archive_path();
+  ###########
+  # set qc_in
```

Comment:
If we give files explicitly, we shoudl not need qc_in. Minor changes to parent check object might be needed.